### PR TITLE
IMG-22 Add Support for JPEG2000 rendering - basic

### PR DESCRIPTION
**Pull request by dcruver**
Modified the NitfRenderer to support JPEG2000 rendering if a plugin is found that will
handle JPEG2000 compression.

@bradh 
@kcwire

### CODE REVIEW

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java [hunk 1]
```diff
@@ -137,6 +148,61 @@ private void renderJPEG(final ImageSegment imageSegment, final Graphics2D target
...
+        final ImageReadParam param = reader.getDefaultReadParam();
+
+        if (ImageRepresentation.MULTIBAND.equals(imageSegment.getImageRepresentation())) {
+            final int[] sourceBands = getSourceBands(imageSegment);
+            param.setSourceBands(sourceBands);
+        }
+
+        processBlocks(imageSegment, (r, c) -> {
+                    Rectangle rect = new Rectangle((int) (c * imageSegment.getNumberOfPixelsPerBlockVertical()),
+                            (int) (r * imageSegment.getNumberOfPixelsPerBlockHorizontal()),
```
Comment (bradh) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: I'm having trouble convincing myself that this logic is right for non-square blocks. Think of an image that is 400 pixels wide by 300 pixels high. Assume each block is 200 pixels wide (horizontal) by 100 pixels high (vertical). So its 2 blocks wide and 3 blocks high. For block (0,0) it doesn't matter. But it looks like (0,1) will start at (100, 0), but I think that should be (200,).  Maybe see if you can find a test case with non-square blocks?

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java [hunk 2]
```diff
@@ -137,6 +148,61 @@ private void renderJPEG(final ImageSegment imageSegment, final Graphics2D target
...
+        );
+    }
+
+    private int[] getSourceBands(final ImageSegment imageSegment) {
+        List<Integer> imageBands = new ArrayList<Integer>();
+
+        for (int i = 0; i < imageSegment.getNumBands(); i++) {
+            ImageBand band = imageSegment.getImageBandZeroBase(i);
+
+            if (null != band.getImageRepresentation()) {
```
Comment (bradh) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: There are some YCbCr format images in the JITC samples. Do you want to try to handle this or just add a ticket for "Future"?

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java [hunk 3]
```diff
@@ -76,6 +84,9 @@ public final void render(final ImageSegment imageSegment, final Graphics2D targe
             ImageMask imageMask = new ImageMask(imageSegment, imageSegment.getData());
             renderJPEG(imageSegment, targetGraphic, imageMask);
             break;
+        case JPEG2000:
```
Comment (bradh) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: Should make a "Future" JIRA ticket for the JPEG2000MASK (M8) case.

Comment (dcruver) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: :+1:

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java (hunk 2 above)
Comment (dcruver) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: So far I haven't run into any examples of that, do let's add a ticket for it.

File: render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java (hunk 1 above)
Comment (dcruver) [inline render/src/main/java/org/codice/imaging/nitf/render/NitfRenderer.java]: I pulled this code out into a test class to run the scenario you outlined.  The x and y calculations had PPBh and PPBv swapped.  Changed this.

**Pull request merged**